tests/console/validate_lvm_encrypt.pm

```diff
@@ -84,13 +152,75 @@ sub run {
+    my $parts = extract_partitions_from_disk($drives);
+    record_info('PARTS', 'Total = ' . scalar @{$parts});
+    my $crypts = get_crypt_containers($parts);
+    record_info('CRYPT', 'Total = ' . scalar @{$crypts});
+    my $encrypted_volumes = get_encrypted_volumes($crypts);
+    record_info('VOLUMES',  'Total = ' . scalar @{$encrypted_volumes});
+    record_info($_->{name}, "Mounted in $_->{mountpoint} ($_->{type})") foreach (@{$encrypted_volumes});
+
+    # read kernel boot parameters, determine root partition
+    my ($rootfs_part) = grep { /root=[^\s]*/ }
+      split(/\s/, script_output qq[cat /proc/cmdline | tee /dev/$serialdev]);
+    # possible formats
+    # root=UUID=aaf272eb-7817-4e04-9f4d-da4e4e354706
+    # root=/dev/mapper/vg+lv
+    # root=/dev/sdaX
+    die "No rootfs found in /proc/cmdline!\n" unless defined($rootfs_part);
+    record_info('rootfs', $rootfs_part = (split(/\=/, $rootfs_part, 2))[1]);
+
+    # get partition which holds initrd, or where does /boot live?
+    my $initrd_record = script_output('df -T /boot/initrd-$(uname -a | cut -f 3 -d\' \') | tail -n 1| tee /dev/' . "$serialdev");
+    my $boot_part     = (split(/\s/, $initrd_record))[0];
+    assert_script_run("df | grep -P \'^$boot_part\'| grep boot");
+    record_info('boot', $boot_part);
+
+    ### rootfs != boot
+    # a) explicitly separated /boot
+    # b) UEFI - old storage stack puts whole /boot on separate partition
+    #         - storage-ng separates only /boot/efi
+    my $is_separated_special_boot;
+    my $is_boot_on_rootfs = eval($boot_part eq $rootfs_part);
+    if (get_var('UEFI')) {
+        my $boot_efi = script_output('df -T /boot/efi| tail -n 1| awk \'{print $1}\' |tee /dev/' . "$serialdev");
+        record_info('EFI', $boot_efi);
+        $is_separated_special_boot = eval($boot_part ne $boot_efi);
+    } elsif (is_s390x) {
+        my $boot_zipl = script_output('df -T /boot/zipl| tail -n 1| awk \'{print $1}\' |tee /dev/' . "$serialdev");
+        record_info('ZIPL', $boot_zipl);
+        $is_separated_special_boot = eval($boot_part ne $boot_zipl);
+    } else {
+        bmwqemu::diag("x86_64 and ppc64le don't have any additional boot partition to be mounted");
     }
-    else {                                                                     # Encrypted boot partition with lvm
-        assert_script_run q[df | grep -P "^\/dev/\mapper\/\w+"| grep boot];
+
+    # all installations can separate /boot
+    if ((get_var('NAME') =~ m/separate/) or
+        ((get_var('NAME') =~ m/cryptlvm/) and !is_storage_ng)) {
+        record_info('Expected', "Dedicated partition for /boot", result => ($is_boot_on_rootfs) ? 'fail' : 'ok');
+    } else {
+        record_info('Expected', "/boot is part of rootfs", result => ($is_boot_on_rootfs) ? 'ok' : 'fail');
+    }
+
+    # handle special boot partition
+    if (get_var('UEFI') || is_s390x || is_aarch64) {
+        record_info('Expected', "/boot/zipl or /boot/efi is located on a separate partition",
```

Comment:
AFAIR, record info with `fail` as a result  won't fail the test suite, but just use red border. We have deployed `Test::Assert` perl module, it looks like perfect use case for it.

Reply:
Yup, but on the otherhand, we have no test data which should come at some point in the future as well. I assumed that we want to verify the configuration against test data from the test suite. But let me check the assert functions.

Reply:
That's not exactly the point. Problem here is if this check fails, it won't be visible on the dashboard, but only when you open the job. I'm not 100% sure, so if I'm wrong then we obviously can keep it the way it is.

Reply:
Ofc, I was just relying on that we are going to add the test data anyway. That will require code change too. But it is already changed and I am pushing the code right now :)